Packages/Tracking Preview/HandRays/Runtime/Scripts/RayInteractors/HandRayInteractor.cs

```diff
@@ -0,0 +1,108 @@
+/******************************************************************************
+ * Copyright (C) Ultraleap, Inc. 2011-2022.                                   *
+ *                                                                            *
+ * Use subject to the terms of the Apache License 2.0 available at            *
+ * http://www.apache.org/licenses/LICENSE-2.0, or another agreement           *
+ * between Ultraleap and you, your company or other organization.             *
+ ******************************************************************************/
+
+using System;
+using UnityEngine;
+
+namespace Leap.Unity.Preview.HandRays
+{
+    /// <summary>
+    /// Abstract base class for HandRay Interactors
+    /// Use this class as a way to define how your hand ray gets cast out
+    /// </summary>
+    public abstract class HandRayInteractor : MonoBehaviour
+    {
+        public HandRay handRay => _handRay;
+        [SerializeField] private HandRay _handRay;
+        
+        [Header("Layer Logic")]
+        public FarFieldLayerManager farFieldLayerManager;
+
+        [Tooltip("Note that the far field object layer will be added automatically from the FarFieldLayerManager")]
+        public LayerMask layerMask = 0;
+        
+        [Tooltip("If true, on start the floor layer referenced in FarFieldLayerManager will get added to the layer mask")]
+        public bool autoIncludeFloorLayer = false;
+
+        /// <summary>
+        /// Called everytime a raycast is fired
+        /// </summary>
+        public Action<RaycastHit[], RaycastHit> OnRaycastUpdate;
+
+        /// <summary>
+        /// The points returned by the raycast, which can be used by a HandRayRenderer
+        /// </summary>
+        [HideInInspector] public Vector3[] linePoints;
+        /// <summary>
+        /// The number of points returned by the raycast
+        /// </summary>
+        [HideInInspector] public int numPoints;
```

Comment:
Seems odd to do this as this info is on the linePoints array.

Reply:
From what I understand, this is how it is used within XRI too

The line renderers reference the interactor to know how to draw their lines, regardless of any input 'rays' as the Interactor is the central source of information